### ISSUE TITLE
GPU accelerated computing with OpenFF

### ISSUE THREAD

**Issue by khuddzu** [openforcefield/openff-toolkit#1191]
Hi on your installation page:
https://open-forcefield-toolkit.readthedocs.io/en/latest/installation.html
It says in windows GPU compute is supported. 
I am using Linux. 
I was wondering how to make your software run on GPU rather than CPU in Linux. Below is my code(you might recognize me and my code from a previous issue. ha sorry for still bothering) just in case this is helpful. 

```
#OpenFF
import openff
import openff.toolkit
from openff.toolkit.topology import Molecule
from openff.toolkit.utils.toolkits import AmberToolsToolkitWrapper
import torch
import torchani
from torchani.io import tensor_to_xyz
import myfunc as mf
import tempfile
#from openbabel import pybel
from rdkit import Chem
from datetime import datetime

ani2x = '/data/khuddzu/ANI-Data/data/ani2x'
a2x_1x = '{}/ANI-1x-wB97X-631Gd.h5'.format(ani2x)
a2x_2x = '{}/ANI-2x_heavy-wB97X-631Gd.h5'.format(ani2x)
a2x_d = '{}/ANI-2x_dimers-wB97X-631Gd.h5'.format(ani2x)
mini = '/data/khuddzu/ANI-Data/mini/mini_ani2x_wb97mv.h5'

#ds = torchani.datasets.ANIDataset(a2x_2x)


ds = torchani.datasets.ANIDataset(locations=(a2x_1x, a2x_2x, a2x_d), names=('1x', '2x', 'dimers'))

for k,v in ds.items():
    species = ds[k]['species']
    coordinates = ds[k]['coordinates']
    for s, c in zip(species, coordinates):                                      #iterating through each conformation
        s = s.unsqueeze(0)
        c = c.unsqueeze(0)
        try:
            rdmol = mf.create_RdMOL_object(s, c, canonicalize=True, rm_H=False)
            mol = Molecule.from_rdkit(rdmol)
            stime = datetime.now()
            mol.assign_partial_charges(partial_charge_method='am1bcc', toolkit_registry=AmberToolsToolkitWrapper())
            etime = datetime.now()-stime
            print(etime)
        except:
            continue


```

**Comment by mattwthompson** [comment]
Hi @khuddzu,

The OpenFF stack generally delegates its heavy lifting to other packages via its wrappers. Some of these packages can probably run on GPUs but that's really up to each one. The line you saw in the docs means to generally refer to whether or not other steps in a workflow can be accelerated with GPUs on Windows, i.e. "I'm on Windows, will OpenMM use my GPU?" As you're on Linux you should be able to take advantage of these speed-ups wherever they're available. The toolkit code probably won't ever run on GPUs itself, both because it's hard to port pure Python code and most of the compute time is spent waiting for other packages to do compute.

In your case most of the CPU time is probably spent in subprocesses calling `antechamber` and waiting for it to assign partial charges. I don't think `antechamber` can run on GPUs, so your options are limited to assigning partial charges with a different method/another tool or looping in parallel with something like [`Dask`](https://dask.org/) or `multiprocessing`. In the future, OpenFF hopes to deliver some alternative(s) to `antechamber` that run much faster and on GPUs, but they're not ready for production use.

**Comment by khuddzu** [comment]
Thank you!!
I might need to open another issue for this, however, I was wondering if you might help me with this. 
Say I want to run the equivalent of this with openff:
```
antechamber -i l.pdb -fi pdb -o l.mol2 -fo mol2 -c bcc -ek “maxcyc=0”
```

I found this part of the assign_partial_charges function:

```
                # Compute desired charges
                # TODO: Add error handling if antechamber chokes
                short_charge_method = charge_method["antechamber_keyword"]
                subprocess.check_output(
                    [
                        "antechamber",
                        "-i",
                        "molecule.sdf",
                        "-fi",
                        "sdf",
                        "-o",
                        "charged.mol2",
                        "-fo",
                        "mol2",
                        "-pf",
                        "yes",
                        "-dr",
                        "n",
                        "-c",
                        short_charge_method,
                        "-nc",
                        str(net_charge),
                    ]
                )
```

If I change it to this: 

```
                # Compute desired charges
                # TODO: Add error handling if antechamber chokes
                short_charge_method = charge_method["antechamber_keyword"]
                subprocess.check_output(
                    [
                        "antechamber",
                        "-i",
                        "molecule.sdf",
                        "-fi",
                        "sdf",
                        "-o",
                        "charged.mol2",
                        "-fo",
                        "mol2",
                        "-pf",
                        "yes",
                        "-dr",
                        "n",
                        **"-ek",
                        '"maxcyc=0"',**
                        "-c",
                        short_charge_method,
                        "-nc",
                        str(net_charge),
                    ]
                )
```
(Changes in bold)
Should I be seeing the same results as the desired calculation? 
I find this might help with my timing issue. Not allowing an optimization in antechamber.
